extension/wasm/wasm_bindings.cpp

```diff
@@ -0,0 +1,561 @@
+      default:
+        THROW_JS_ERROR(TypeError, "Unsupported Tensor type: %d", scalar_type);
+    }
+  }
+
+  static std::unique_ptr<JsTensor> from_iter(
+      val_array<int> sizes,
+      val_array<val> data,
+      val type = val::undefined(),
+      val_array<int> dim_order = val::undefined(),
+      val_array<int> strides = val::undefined()) {
+    auto sizes_vec =
+        convertJSArrayToNumberVector<executorch::aten::SizesType>(sizes);
+
+    auto dim_order_vec = dim_order.isUndefined()
+        ? std::vector<executorch::aten::DimOrderType>()
+        : convertJSArrayToNumberVector<executorch::aten::DimOrderType>(
+              dim_order);
+    auto strides_vec = strides.isUndefined()
+        ? std::vector<executorch::aten::StridesType>()
+        : convertJSArrayToNumberVector<executorch::aten::StridesType>(strides);
+
+    // If type is undefined, infer the type from the data.
+    // Assume it is a Bigint if not Number.
+    ScalarType scalar_type = type.isUndefined()
+        ? (data["length"].as<size_t>() == 0 || data[0].isNumber()
+               ? ScalarType::Float
+               : ScalarType::Long)
+        : type.as<ScalarType>();
+    switch (scalar_type) {
+#define JS_CASE_FROM_ITER_VECTOR_TYPE(T, NAME)                 \
+  case ScalarType::NAME: {                                     \
+    auto data_vec = convertJSGeneratorToNumberVector<T>(data); \
+    assert_valid_numel(data_vec, sizes_vec);                   \
+    TensorPtr tensor = make_tensor_ptr(                        \
+        std::move(sizes_vec),                                  \
+        std::move(data_vec),                                   \
+        std::move(dim_order_vec),                              \
+        std::move(strides_vec),                                \
+        ScalarType::NAME);                                     \
+    return std::make_unique<JsTensor>(std::move(tensor));      \
+  }
+      JS_FORALL_SUPPORTED_TENSOR_TYPES(JS_CASE_FROM_ITER_VECTOR_TYPE)
+      default:
+        THROW_JS_ERROR(TypeError, "Unsupported Tensor type: %d", scalar_type);
+    }
+  }
+
+ private:
+  TensorPtr tensor_;
+};
+
+// Converts JS value to EValue.
+EValue to_evalue(val v) {
+  if (v.isUndefined()) {
+    THROW_JS_ERROR(TypeError, "Value cannot be undefined");
+  }
+  if (v.isNull()) {
+    return EValue();
+  } else if (v.isNumber()) {
+    return EValue(v.as<double>());
+  } else if (v.isTrue()) {
+    return EValue(true);
+  } else if (v.isFalse()) {
+    return EValue(false);
+  } else {
+    const std::string& type_str = v.typeOf().as<std::string>();
+    if (type_str == "bigint") {
+      return EValue(v.as<int64_t>());
+    } else if (type_str == "object") {
+      // If it is an object, assume it is a tensor.
+      return EValue(v.as<JsTensor&>().get_tensor());
+    }
+    THROW_JS_ERROR(
+        TypeError, "Unsupported JavaScript type: %s", type_str.c_str());
+  }
+}
+
+// Converts EValue to JS value.
+val to_val(EValue v) {
```

Comment:
How about changing the argument the const reference `const EValue& v` or r-value `EValue&& v`